feat: Enable Github Pages hosting Standard Javadocs

alicejli opened the pull request:
Opening a PR for initial review and feedback on https://alicejli.github.io/java-cloud-bom/docs/

Please feel free to leave any comments on the site itself in this PR, or to review the PR directly. 

Things still to be done/improved:

- https://github.com/googleapis/google-auth-library-java/ is missing as it needs a new release for the Javadoc generation to work
- If a new module is created in https://github.com/googleapis/google-cloud-java/, the site requires a manual update to add   in a row for it
- The product and client library links were seeded with a one-time script from the .repo-metadata.json files. Ideally this would be automated to pull a fresh link with each release.

suztomo commented:
I added feedback in the red marker below.

![image](https://github.com/googleapis/java-cloud-bom/assets/28604/92b22cb1-62d6-4f3a-b5d6-71d93795b569)

- libraries-bom is computer-friendly artifact ID for human-friendly name Google Cloud Libraries BOM (https://cloud.google.com/java/docs/bom). This page is for human.
- There's no "Google Cloud Javadoc". It's reference page. Then no need to have "standard" in "Standard Javadoc"
- 

With repeat of "Standard Javadocs | Google Cloud Javadocs | Google Cloud Product Docs" look redundant. I don't know how to make it better. Is it possible to replace "Google Cloud Product Docs" with actual product name with the link?

lqiu96 commented:
Thanks for creating this! I think it looks great and definitely will help Java devs!

A few points of feedback though I might be missing context/ certain features might be a limitation of the javadoc tools/ better implemented elsewhere.

## Documentation Section
![image](https://github.com/googleapis/java-cloud-bom/assets/6621793/05376914-fdd8-4640-9672-a6ceaa6e79f5)
This section currently is empty. Is this going to be populated with other versions once we start releasing with this? It would be nice to be able to select the version of the libraries-bom.

## Library versions
![image](https://github.com/googleapis/java-cloud-bom/assets/6621793/a05c9089-eca3-45e6-8aa9-fdd1fc5f8745)
Would it be possible to list library versions next to the library itself? I have to click into the Javadocs link to find the version: 
![image](https://github.com/googleapis/java-cloud-bom/assets/6621793/6513d001-a03b-41e0-93ef-feed14fe2719)

## Ordering
![image](https://github.com/googleapis/java-cloud-bom/assets/6621793/60a12119-41cb-4813-88e0-d2841a3265eb)
`google-cloud-dlp` seems to be out of place -- not sure why it's listed at the bottom of the page. 

Also, is it possible to organize by "type" of library? One section for like runtime libraries (gax, api-common) and another section for product libraries (aiplatform, compute, etc.).

## Handwritten libraries (Bom vs No-Bom)
![image](https://github.com/googleapis/java-cloud-bom/assets/6621793/35463f07-36fa-4c36-b6a8-21bb004c6a1f)
Some libraries have `-bom` and others don't (bigquery vs bigtable-bom). Maybe this is intended, but thought it was inconsistent with the GAPIC libraries.

alicejli commented:
@suztomo Thank you for your review! Comments inline.

> * libraries-bom is computer-friendly artifact ID for human-friendly name Google Cloud Libraries BOM (https://cloud.google.com/java/docs/bom). This page is for human.

I see your point. I want to keep the `libraries-bom` artifact ID to match the other artifact IDs for consistency, but I've added a new topline header that says `Google Cloud Libraries BOM`, and the `libraries-bom` line is smaller. Let me know what you think.

> * There's no "Google Cloud Javadoc". It's reference page. Then no need to have "standard" in "Standard Javadoc"

So right now the Google Cloud Reference pages have the exact same content as the Javadocs, just in a separate format. I wanted to convey the fact that they show the same content, just in different formats to reduce confusion. However, I do agree that the term "Google Cloud Javadoc" is confusing, so I've updated it to read "Google Cloud Library Reference."

> * The repeat of "_Standard Javadocs | Google Cloud Javadocs | Google Cloud Product Docs_" look redundant. I don't know how to make it better. 

I took Lawrence's idea below, and changed it to read the version of the artifact. I think it looks neater; let me know what you think.

> * Is it possible to replace "Google Cloud Product Docs" with actual product name with the link?

Updated to read the `name_pretty` field from the `.repo-metadata.json` file.

alicejli commented:
@lqiu96 Thank you for your review! Comments in-line.

> ## Documentation Section
> This section currently is empty. Is this going to be populated with other versions once we start releasing with this? It would be nice to be able to select the version of the libraries-bom.

Yeah so that was the original plan except Github Pages has a 2GB limit for their sites, so unfortunately we can only comfortable host the latest version of the libraries-bom. I kept the formatting as-is as ideally we do want to eventually host historical versions, but we would need to look into other hosting options for that.


> ## Library versions
> Would it be possible to list library versions next to the library itself? I have to click into the Javadocs link to find the version

Great idea - I've combined your feedback with Tomo's comment above and added in the version as the link to the Javadocs / GCP Library Reference.


> ## Ordering
`google-cloud-dlp` seems to be out of place -- not sure why it's listed at the bottom of the page.

Good catch; I fixed a bug in my alphabetizing of the table so that should be fixed now.
 
> Also, is it possible to organize by "type" of library? One section for like runtime libraries (gax, api-common) and another section for product libraries (aiplatform, compute, etc.).

I like that idea - I'll look into adding another column for "type." Stay tuned.

> ## Handwritten libraries (Bom vs No-Bom)
> Some libraries have `-bom` and others don't (bigquery vs bigtable-bom). Maybe this is intended, but thought it was inconsistent with the GAPIC libraries.

The handwritten libraries are pulling directly from the `google-cloud-bom/pom.xml` file for the dependencies, and are consistent with the artifact_id listed there.

suztomo commented:
The site looks great https://alicejli.github.io/java-cloud-bom/docs/ Thank you for incorporating feedback.

The files changed show "♾️". It seems something went wrong with Git operation.

meltsufin commented:
It looks like the best Java SDK documentation index we have today!

lqiu96 commented:
Thanks for the info! Site looks good to me :) 

> Yeah so that was the original plan except Github Pages has a 2GB limit for their sites, so unfortunately we can only comfortable host the latest version of the libraries-bom. I kept the formatting as-is as ideally we do want to eventually host historical versions, but we would need to look into other hosting options for that.

Makes sense. Do you this is something worth noting somewhere on the site (maybe like a footnote or disclaimer)?

> The handwritten libraries are pulling directly from the `google-cloud-bom/pom.xml` file for the dependencies, and are consistent with the artifact_id listed there.

Ah gotcha. Ideally they're all consistent, but I think this is fine.

alicejli commented:
Please see https://github.com/googleapis/java-cloud-bom/pull/6069 for a pared down version (doesn't include all the static generated files)
